drivers/ethernet/eth_stm32_hal.c

```diff
@@ -845,7 +849,26 @@ static int eth_stm32_hal_set_config(const struct device *dev,
 		net_if_set_link_addr(dev_data->iface, dev_data->mac_addr,
 				     sizeof(dev_data->mac_addr),
 				     NET_LINK_ETHERNET);
-		return 0;
+		ret = 0;
+		break;
+	case ETHERNET_CONFIG_TYPE_PROMISC_MODE:
+#if defined(CONFIG_NET_PROMISCUOUS_MODE)
+               if(config->promisc_mode) {
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+		heth->Instance->MACPFR |= ETH_MACPFR_PR;
+#else
+		heth->Instance->MACFFR |= ETH_MACFFR_PM;
+#endif  /* CONFIG_SOC_SERIES_STM32H7X */
+               } else {
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+		heth->Instance->MACPFR &= ~ETH_MACPFR_PR;
+#else
+		heth->Instance->MACFFR &= ~ETH_MACFFR_PM;
+#endif  /* CONFIG_SOC_SERIES_STM32H7X */
+               }
```

Comment:
The following would be more readable:

```suggestion
#if defined(CONFIG_NET_PROMISCUOUS_MODE)
#if defined(CONFIG_SOC_SERIES_STM32H7X)
               if(config->promisc_mode) {
               		heth->Instance->MACPFR |= ETH_MACPFR_PR;
               } else {
               		heth->Instance->MACPFR &= ~ETH_MACPFR_PR;
               }
#else
                if(config->promisc_mode) {
               		 heth->Instance->MACFFR |= ETH_MACFFR_PM;
               } else {
               		heth->Instance->MACFFR &= ~ETH_MACFFR_PM;
               }
#endif  /* CONFIG_SOC_SERIES_STM32H7X */
#endif /* CONFIG_NET_PROMISCUOUS_MODE */
```

Reply:
Agree with it. Thanks. Yesterday it was late night, as I did testings and changes.

Reply:
@Nukersson that's why we're multiple eyeballs in reviews. Thanks for your valuable review anyway.